src/Microsoft.AspNet.Mvc.Core/Filters/DefaultFilterProvider.cs

```diff
@@ -31,6 +34,10 @@ public void OnProvidersExecuting([NotNull] FilterProviderContext context)
                 {
                     ProvideFilter(context, item);
                 }
+                context.Results =
+                    context.Results
+                    .OrderBy(filter => filter.Descriptor, FilterDescriptorCorsComparer.Comparer)
```

Comment:
Also wasn't in design iterations, but would also suggest just giving these filters a high priority order and not hardcoding this behavior. What you're doing here is **not** a guarantee, the user can add more filter providers. 

We might even have cases where someone very smart is trying to work around this behavior and has to struggle with an extra layer of 'security'.